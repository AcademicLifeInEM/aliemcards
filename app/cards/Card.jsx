/* eslint-disable react/no-danger */
import React, { PureComponent, PropTypes } from 'react';
import { post } from 'axios';
import marked from 'marked';

export default class Card extends PureComponent {

  static propTypes = {
    match: PropTypes.shape({
      params: PropTypes.shape({ id: PropTypes.string }),
    }).isRequired,
  }

  constructor(props) {
    super(props);
    this.state = {
      authors: [],
      categories: [],
      content: '',
      created: null,
      id: '',
      title: '',
      updates: [],
      lastUpdate: '',
    };
  }

  componentDidMount() {
    this.getCard(this.props.match.params.id);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.match.params.id === this.props.match.params.id) {
      return;
    }
    this.getCard(nextProps.match.params.id);
  }

  getCard(cardId) {
    post('/graphql', {
      query: `query getCardById($id: String!) {
        card(id: $id) {
          id
          title
          authors {
            name
          }
          created
          updates
          content
        }
      }`,
      variables: { id: cardId },
    })
    .then(res => {
      if (res.status !== 200) throw res.status;
      const { card } = res.data.data;
      const lastUpdate = card.updates
        ? new Date(card.updates[0]).toLocaleDateString('en-US')
        : new Date(card.created).toLocaleDateString('en-US');
      this.setState({
        ...card,
        lastUpdate,
      });
    })
    .catch(err => console.error(`Error: API response status code = ${err}`));
  }

  getContent() {
    return {
      __html: marked(this.state.content),
    };
  }

  render() {
    return (
      <article role="article">
        <h1>{this.state.title}</h1>
        <div className="card">
          <div className="card__meta">
            <div>
              <strong>{this.state.authors.length > 1 ? 'Authors: ' : 'Author: '}</strong>
              {this.state.authors.map(author => author.name).join(', ')}
            </div>
            <div><strong>Updated:</strong> {this.state.lastUpdate}</div>
          </div>
          <div className="card__content" dangerouslySetInnerHTML={this.getContent()} />
        </div>
      </article>
    );
  }
}
