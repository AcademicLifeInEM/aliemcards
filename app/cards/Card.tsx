import * as React from 'react';
import { gql, graphql } from 'react-apollo';
import { Helmet } from 'react-helmet';
import * as remark from 'remark';
import * as html from 'remark-html';

import { Card as ICard } from '../../server/schema';
import AddThis from '../components/AddThis';

interface Props {
  data: {
    card: ICard;
    networkStatus: number;
  };
  match: {
    params: {
      id: string;
    };
  };
  location: {
    pathname: string;
  };
}

const cardDataFromId = gql`
  query getCardById($id: String!) {
    card(id: $id) {
      id
      title
      authors {
        name
      }
      created
      updates
      content
    }
  }
`;

const config = {
  options: ({ match }) => ({ variables: { id: match.params.id } }),
};

@graphql(cardDataFromId, config)
export default class Card extends React.Component<Props> {
  render() {
    const { card, networkStatus } = this.props.data;
    if (networkStatus < 7) {
      return <div>Loading...</div>;
    }
    const lastUpdate = card.updates
      ? new Date(card.updates[0]).toLocaleDateString('en-US', {
          timeZone: 'UTC',
        })
      : new Date(card.created).toLocaleDateString('en-US', {
          timeZone: 'UTC',
        });
    return (
      <article role="article">
        <Helmet>
          <script type="application/ld+json">{JSON.stringify({ headline: card.title })}</script>
        </Helmet>
        <AddThis path={this.props.location.pathname} title={card.title} />
        <h1>{card.title}</h1>
        <div className="card">
          <div className="card__meta">
            <div>
              <strong>{card.authors.length > 1 ? 'Authors: ' : 'Author: '}</strong>
              {card.authors.map(author => author.name).join(', ')}
            </div>
            <div>
              <strong>Updated:</strong> {lastUpdate}
            </div>
          </div>
          <div
            className="card__content"
            dangerouslySetInnerHTML={{
              __html: remark()
                .use(html)
                .processSync(card.content).contents,
            }}
          />
        </div>
      </article>
    );
  }
}
