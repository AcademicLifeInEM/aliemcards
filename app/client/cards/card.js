import React from 'react';
import { post } from 'axios';
import marked from 'marked';

marked.setOptions({
  renderer: new marked.Renderer(),
  gfm: true,
  tables: true,
  breaks: false,
  pedantic: false,
  sanitize: false,
  smartLists: true,
  smartypants: false,
});

export default class Card extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      authors: [],
      categories: [],
      content: '',
      created: null,
      drugs: null,
      id: '',
      title: '',
      updates: [],
      lastUpdate: '',
    };
  }

  componentDidMount() {
    console.log(this.props);
    post('/graphql', {
      query: `query getCardById($id: ID!) {
        card(id: $id) {
          id
          title
          authors
          created
          updates
          content
        }
      }`,
      variables: { id: this.props.params.id },
    })
    .then(res => {
      if (res.status !== 200) throw res.status;
      const { card } = res.data.data;
      const lastUpdate = card.updates
        ? new Date(card.updates[0]).toLocaleDateString('en-US')
        : new Date(card.created).toLocaleDateString('en-US');
      this.setState({
        ...card,
        lastUpdate,
      });
    })
    .catch(err => console.error(`Error: API response status code = ${err}`));
  }
  getContent() {
    return {
      __html: marked(this.state.content),
    };
  }

  render() {
    return (
      <div>
        <h1>{this.state.title}</h1>
        <ul className="cardMeta">
          <li><AuthorList authorArray={this.state.authors} /></li>
          <li><strong>Updated:</strong> {this.state.lastUpdate}</li>
        </ul>
        <div className="cardHtml" dangerouslySetInnerHTML={this.getContent()} />
      </div>
    );
  }
}

Card.propTypes = {
  params: React.PropTypes.shape({
    id: React.PropTypes.string,
  }),
};

const AuthorList = ({ authorArray }) =>
  <span>
    <strong>{authorArray.length > 1 ? 'Authors' : 'Author'}: </strong>
    {authorArray.map(author => <span key={author} className="author">{author}</span>)}
  </span>;

AuthorList.propTypes = {
  authorArray: React.PropTypes.arrayOf(String),
};
AuthorList.defaultProps = {
  authorArray: [],
};
