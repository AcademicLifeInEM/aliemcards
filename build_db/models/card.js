const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const cardSchema = new Schema({
  title: String,
  slug: { type: String, required: true, unique: true, index: true },
  tags: Array,
  categories: Array,
  content: String,
  hash: String,
  createdAt: Date,
  updatedAt: Date,
});

cardSchema.pre('save', (next) => {
  const currentDate = new Date();
  this.updatedAt = currentDate;
  if (!this.created_at) this.createdAt = currentDate;
  next();
});

module.exports = mongoose.model('Card', cardSchema);
