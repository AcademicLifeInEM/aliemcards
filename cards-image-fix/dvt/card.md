---

title: 'DVT of the Leg - ACCP Guidelines'
authors:
    - 'Michelle Lin, MD'
created: 2013/01/24
updates: null
categories:
    - Cardiovascular
    - EBM
drugs: null

---



# Deep Vein Thrombosis (DVT) of Leg: 2012 ACCP Guidelines

**Wells Clinical Prediction Criteria for DVT (Determine pretest probability)**

|                                                    |     |
|----------------------------------------------------|-----|
| Active malignancy                                  | +1  |
| Major surgery within 4 weeks or bedridden &gt; 3 days   | +1  |
| Asymmetric calf swelling &gt; 3 cm in affected leg | +1  |
| Collateral, non-varicose superficial veins         | +1  |
| Entire leg swelling                                | +1  |
| Localized tenderness along deep venous system      | +1  |
| Pitting edema greater in symptomatic leg           | +1  |
| Paralysis, paresis, or recent leg immobilization   | +1  |
| Previous DVT                                       | +1  |
| Alternative diagnosis equally or more likely       | -2  |

**Apply Wells score** to the [2012 ACCP Evidence-Based Clinical Practice Guideline](http://www.ncbi.nlm.nih.gov/pubmed/22315267) as summarized below.

## Wells score &lt; 1 (LOW pretest probability, 5%) 

![](image-1.png)

## Wells Score 1-2 (Moderate pretest probability, 17%)  

***Two options:*** start with D-dimer or ultrasound 

***Note:*** If significant comorbidities exist that may cause D-Dimer elevation, start with ultrasound as initial test. If ultrasound is unfeasible (excessive tissue or leg casting), use venography.

![](image-2.png)

## Wells score &gt; 2 (High pretest probability, 53%) 

Do NOT begin with D-Dimer. Follow Option \#2 from Moderate Pretest Probability.

![](image-3.png)

## Pregnant patients 

![](image-4.png)

***Note:*** If clinical suspicion of iliac vein thrombosis (entire leg swelling), proceed to iliac vein U/S or venography, if initial proximal leg U/S is negative.

## References

-   [Bates SM, Jaeschke R, Stevens SM, Goodacre S, Wells PS, Stevenson MD, Kearon C, Schunemann HJ, Crowther M, Pauker SG, Makdissi R, Guyatt GH; American College of Chest Physicians.Diagnosis of DVT: Antithrombotic Therapy and Prevention of Thrombosis, 9th ed: American College of Chest Physicians Evidence-Based Clinical Practice Guidelines. Chest. 2012 Feb;141(2 Suppl):e351S-418S.](http://www.ncbi.nlm.nih.gov/pubmed/22315267)
