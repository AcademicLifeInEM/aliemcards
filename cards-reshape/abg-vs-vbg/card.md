


# ABG versus VBG

## ABG

-   **PROS**: Tells you exactly what the arterial environment looks like (pH, PaCO2, HCO3)
-   **CONS**: Painful, increased risk of bleeding, hematoma, infection, nerve injury, digitalischemia, injury to staff, delays in care, often requires serial exams

## **VBG**

<table>
<thead>
<tr class="header">
<th><br />
</th>
<th><strong>Correlation with ABG</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>pH</strong></td>
<td><ul>
<li>Good correlation within 0.03-0.04 units [1-4]<br />
</li>
</ul></td>
</tr>
<tr class="even">
<td><strong>pCO2</strong></td>
<td><ul>
<li>Good correlation [4], although may be non-correlative in severe shock [3]</li>
<li>100% sensitive in detecting arterial hypercarbia in COPD exacerbations using cutoff of PaCO2 45 mmHg [1]</li>
<li>Mean difference pCO2 -4 (-12.7 - 4.7 <em>[95%CI]</em>) [4]</li>
</ul></td>
</tr>
<tr class="odd">
<td><strong>HCO3</strong></td>
<td><ul>
<li>Good correlation</li>
<li>Mean difference 0.52 (-1.81 - 2.85 <em>[95%CI]</em>) - 0.9 (-3.35 - 5.09 <em>[95% CI]</em>) [3-4]</li>
</ul></td>
</tr>
<tr class="even">
<td><strong>Lactate</strong></td>
<td><ul>
<li>Good correlation</li>
<li>Mean difference 0.08 (-0.27 - 0.42 <em>[95%CI]</em>) [3]</li>
</ul></td>
</tr>
<tr class="odd">
<td><strong>Base<br />
excess</strong></td>
<td><ul>
<li>Good correlation</li>
<li>Mean difference 0.19 (-1.86 - 2.24 <em>[95%CI]</em>) [3]</li>
</ul></td>
</tr>
</tbody>
</table>

**pH assessment in DKA:**

-   VBG correlated with ABG well
-   Mean difference -0.015 ± 0.006 units *\[95% CI\]*
-   ABG pH changed treatment or disposition in 2.5% cases compared to VBG pH

**CONS:**
-   May still require ABG if pCO2 is ≥ 45 mm Hg or severe shock

## ALGORITHM FOR ORDERING VBG vs. ABG

![](image-0.png)

## References

-   [1. McCanny P, Bennett K, Staunton P, McMahon G. Venous vs arterial blood gases in the assessment of patients presenting with an exacerbation of chronic obstructive pulmonary disease. Am J Emerg Med. 2012 Jul;30(6):896-900.](https://www.ncbi.nlm.nih.gov/pubmed?term=21908141)
-   [2. Kelly AM, McAlpine R, Kyle E. Venous pH can safely replace arterial pH in the initial evaluation of patients in the emergency department. Emerg Med J. 2001 Sep;18(5):340-2](https://www.ncbi.nlm.nih.gov/pubmed?term=11559602)
-   [3. Middleton P, Kelly AM, Brown J, Robertson M. Agreement between arterial and central venous values for pH, bicarbonate, base excess, and lactate. Emerg Med J. 2006 Aug;23(8):622-4](https://www.ncbi.nlm.nih.gov/pubmed?term=16858095)
-   [4. Koul PA, Khan UH, Wani AA, Eachkoti R, Jan RA, Shah S, Masoodi Z, Qadri SM, Ahmad M, Ahmad A. Comparison and agreement between venous and arterial gas analysis in cardiopulmonary patients in Kashmir valley of the Indian subcontinent. Ann Thorac Med. 2011 Jan;6(1):33-7.](https://www.ncbi.nlm.nih.gov/pubmed?term=21264169)
-   [5. Ma OJ, Rush MD, Godfrey MM, Gaddis G. Arterial blood gas results rarely influence emergency physician management of patients with suspected diabetic ketoacidosis. Acad Emerg Med. 2003 Aug;10(8):836-41](https://www.ncbi.nlm.nih.gov/pubmed?term=12896883)
