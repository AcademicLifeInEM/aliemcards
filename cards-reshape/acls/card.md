


# ACLS 2010 Algorithm for Cardiac Arrest

![](image-0.png)

**Tips:**

-   **2 minutes** of continuous, uninterrupted CPR is key before rechecking the rhythm.
-   Think about giving <span class="drug">epinephrine</span> every **OTHER** time you check the rhythm. This puts it at roughly a q4 minute dosing (recommended q3-5 minutes).
-   Don't delay CPR when you recognize a cardiac arrest. That means do this first, and then work on giving oxygen, placing the patient on various monitors, setting up the defibrillator, and establishing IV/IO access. Hence, the C-A-B mnemonic (Circulation before Airway)
-   Chest compressions should be "hard and fast" -- Depth of at least 2 inches and ≥ 100 compressions with a target pCO2 on the end-tidal capnography of ≥10 mm Hg.
-   Use your mobile phone's timer to help alert you every time when 2 minutes is up.

**Additional reversible causes** (not listed on ACLS AHA guidelines):

-   Hypoglycemia
-   Trauma

## Patwari Academy video tutorials

-   [ACLS: Airway](http://academiclifeinem.blogspot.com/2012/11/patwari-academy-videos-acls-and-airway.html)
-   [ACLS: Cardiac arrest](http://academiclifeinem.blogspot.com/2012/12/patwari-academy-videos-acls-parts-4-6.html)
-   [ACLS: Bradycardia and tachycardia](http://academiclifeinem.blogspot.com/2012/12/patwari-academy-videos-acls-parts-7-10.html)

<!-- -->

## References

-   [Neumar RW, et al. Part 8: adult advanced cardiovascular life support: 2010 AHA Guidelines for CPR and Emergency Cardiovascular Care. Circulation. 2010 Nov 2;122(18 Suppl 3):S729-67.](https://www.ncbi.nlm.nih.gov/pubmed/?term=20956224)
