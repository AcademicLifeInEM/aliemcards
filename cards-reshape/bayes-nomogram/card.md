


# Bayes Nomogram

Draw a line connecting your pre-test probability and your likelihood ratio. This provides you with your post-test probability.

![](image-0.png)
