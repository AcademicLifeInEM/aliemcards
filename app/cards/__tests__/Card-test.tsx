jest.mock('react-apollo');

import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Card as CardType } from '../../../server/models/card/cardType';
import Card from '../Card';

const stub: any = {};

const setup = (networkStatus = 7, cardData?: Partial<CardType>) => {
  const card: any = {
    authors: [
      {
        id: '1',
        name: 'John Doe',
        cards: [],
      },
    ],
    categories: [
      {
        id: '1',
        name: 'test',
      },
    ],
    content: '# Hello World',
    created: 1483228800000, // ​​​​​Sun, 01 Jan 2017 00:00:00 GMT​​​​​
    id: '1',
    title: 'Test Card',
    updates: null,
    ...cardData,
  };
  const snapshot = renderer.create(<Card data={{ card, networkStatus }} match={stub} />);
  return {
    snapshot,
  };
};

describe('<Card />', () => {
  it('should render according to networkStatus', () => {
    const { snapshot: loading } = setup(1);
    expect(loading.toJSON()).toMatchSnapshot();

    const { snapshot: polling } = setup(6);
    expect(polling.toJSON()).toMatchSnapshot();

    const { snapshot: rendered } = setup();
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  it('should render with multiple authors and updated date', () => {
    const card: Partial<CardType> = {
      authors: [
        {
          id: '1',
          name: 'John Doe',
          cards: [],
        },
        {
          id: '2',
          name: 'Jane Doe',
          cards: [],
        },
      ],
      updates: [1483228800000],
    };
    const { snapshot } = setup(7, card);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
