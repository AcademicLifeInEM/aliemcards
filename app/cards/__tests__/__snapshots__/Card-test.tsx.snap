// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render according to networkStatus 1`] = `
<div>
  Loading...
</div>
`;

exports[`<Card /> should render according to networkStatus 2`] = `
<div>
  Polling!
</div>
`;

exports[`<Card /> should render according to networkStatus 3`] = `
<article
  role="article"
>
  <h1>
    Test Card
  </h1>
  <div
    className="card"
  >
    <div
      className="card__meta"
    >
      <div>
        <strong>
          Author: 
        </strong>
        John Doe
      </div>
      <div>
        <strong>
          Updated:
        </strong>
         
        1/1/2017
      </div>
    </div>
    <div
      className="card__content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h1>Hello World</h1>
        ",
        }
      }
    />
  </div>
</article>
`;

exports[`<Card /> should render with multiple authors and updated date 1`] = `
<article
  role="article"
>
  <h1>
    Test Card
  </h1>
  <div
    className="card"
  >
    <div
      className="card__meta"
    >
      <div>
        <strong>
          Authors: 
        </strong>
        John Doe, Jane Doe
      </div>
      <div>
        <strong>
          Updated:
        </strong>
         
        1/1/2017
      </div>
    </div>
    <div
      className="card__content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h1>Hello World</h1>
        ",
        }
      }
    />
  </div>
</article>
`;
